### DISCUSSION TITLE
TX2 OTA full disk flashing?

### DISCUSSION THREAD

**Discussion by zach-welch-aquabyte** [OE4T/meta-tegra#972] [Q&A]
I am working with a TX2-based product that currently runs an older version of Yocto built using `meta-tegra`. For various reasons, we want to update the entire eMMC in a device with a new image; however, our deployed devices are all but physically inaccessible, making the normal USB flashing scheme extremely impractical and expensive.

To avoid having to retrieve and crack them open for reprogramming, I want to use a script to pivot the running OS into a `tmpfs`, allowing a new disk image to be downloaded and written to the eMMC on the fly before rebooting. After some digging, I realize that the Yocto build process does not create a suitable image that can be written directly in this manner, as the flashing scripts assemble the pieces on the fly to send via USB.

Thus, I am searching for a method to generate a suitable image as a file on my development host. While I am not seeing a simple path to accomplish that goal for the TX2, I found a method for [cloning a TX1](https://elinux.org/Jetson/TX1_Cloning) that might be adapted to obtain an image from a freshly flashed device, but the language on that page implies that the resulting image will not be suitable for upgrading a fleet of devices in the manner that I require.

Is there a documented process that I could follow to generate the required images, either directly via `bitbake` or indirectly by using the `tegraflash` archive that it produces? Or is USB flashing the only viable means to reflash the entire device?

**Comment by dwalkes** [comment]
@zach-welch-aquabyte we've had some discussions about this in the past, and there are a few pieces implemented which might be useful.  I'd suggest https://github.com/OE4T/meta-tegra/wiki/Over-the-air-reflashing-process and the links there as a jumping off point.  Others might chime in here who have worked on this.
